docs/reference/commandline/run.md

```diff
@@ -242,27 +242,51 @@ For the `overlay2` storage driver, the size option is only available if the
 backing fs is `xfs` and mounted with the `pquota` mount option.
 Under these conditions, user can pass any size less than the backing fs size.
 
-### Mount tmpfs (--tmpfs)
+### Mount tmpfs (--mount type=tmpfs)
+
+> **Note**: The `--tmpfs` flag still works, but `--mount type=tmpfs` is
+> preferred.
 
 ```bash
-$ docker run -d --tmpfs /run:rw,noexec,nosuid,size=65536k my_image
+$ docker run --rm -it \
+  --mount type=tmpfs,tmpfs-size=65536k,tmpfs-mode=600,dst=/run \
+  ubuntu:latest
 ```
 
-The `--tmpfs` flag mounts an empty tmpfs into the container with the `rw`,
-`noexec`, `nosuid`, `size=65536k` options.
+This example mounts a `tmpfs` filesystem 65536k in size into `/run` inside the
+container, with mode 600 (`rw-------`). Because the type of `tmpfs`, the file's
+contents of the file are not written to the container and are only written to a
+block device on the Docker host if memory is low enough to trigger swap usage.
+
+### Mount volume (--mount, -v)
+
+The `--mount` flag allows you to mount volumes, host-directories and `tmpfs`
+mounts in a container.
+
+The `--mount` syntax is preferred. See
+[Differences between `-v` and `--mount` behavior](https://docs.docker.com/storage/volumes/#differences-between--v-and---mount-behavior)
+for more information. These examples show both syntax choices.
+
+#### Bind-mount the current working directory
 
-### Mount volume (-v, --read-only)
+```bash
+$ docker run --mount src=`pwd`,dst=`pwd` -w `pwd` -i -t  ubuntu pwd
+```
 
 ```bash
 $ docker  run  -v `pwd`:`pwd` -w `pwd` -i -t  ubuntu pwd
 ```
 
-The `-v` flag mounts the current working directory into the container. The `-w`
+This example mounts the current working directory into the container. The `-w`
 lets the command being executed inside the current working directory, by
 changing into the directory to the value returned by `pwd`. So this
 combination executes the command using the container, but inside the
 current working directory.
 
+#### Populate the Docker host with the contents of a directory in the  container
```

Comment:
The `-v <host-path>:<container-path>` syntax only creates the _directory_ on the host, but not copy the content from the container to the host. If the container has content at the given location, but the path on the host wasn't found, docker creates an empty directory on the host, and bind mounts that directory over the path in the container.